OSSM-4853 OSSMC Documentation GA (2.5 Release) (#65700)
openshift/openshift-docs: modules/ossm-kiali-ossmc-plugin-user-guide.adoc

// Module included in the following assemblies:
//
//* *service_mesh/v2x/ossm-kiali-ossmc-plugin.adoc

:_mod-docs-content-type: CONCEPT
[id="ossm-kiali-ossmc-plugin-user-guide_{context}"]
= About the OpenShift Service Mesh Console plugin
//In the title include nouns or noun phrases that are used in the body text.
//Do not start the title of concept modules with a verb.

The {SMProductName} Console (OSSMC) plugin is an extension to the {product-title} web console that provides visibility into your Service Mesh. 

[WARNING]
====

> The {SMPRoductName} Console (OSSMC) plugin only supports a single Kiali instance. Whether that Kiali instance is configured to access only a subset of OpenShift projects or has access cluster-wide to all projects does not matter, however, only a single Kiali instance can be accessed.

Reviewer: ```suggestion
The {SMPRoductName} Console (OSSMC) plugin only supports a single Kiali instance. Whether that Kiali instance is configured to access only a subset of OpenShift projects or has access cluster-wide to all projects does not matter. However, only a single Kiali instance can be accessed.
```
I think punctuation is needed to avoid a run-on sentence.

====

You can install the OSSMC plugin in only one of two ways: using the {product-title} web console, or through the CLI.

Installing the OSSMC plugin creates a new category, *Service Mesh*, in the main {product-title} web console navigation. Click *Service Mesh* to see:

* *Overview* for a summary of your mesh displayed as cards that represent the namespaces in the mesh. 

> * *Graph* for a full topology view of your mesh represented by nodes and edges - each node representing a component of the mesh and each edge representing traffic flowing through the mesh between components.

Reviewer: ```suggestion
* *Graph* for a full topology view of your mesh represented by nodes and edges, each node representing a component of the mesh and each edge representing traffic flowing through the mesh between components.
```

Installing the OSSMC plugin creates a new category, *Service Mesh*, in the main OpenShift Console navigation. Click on *Service Mesh* to see:

* *Overview* for a summary of your mesh displayed as cards that represent the namespaces in the mesh. 
* *Graph* for a full topology view of your mesh represented by nodes and edges - each node representing a component of the mesh and each edge representing traffic flowing through the mesh between components.
* *Istio config* for a list of all Istio configuration files in your mesh with a column that provides a quick way to know if the configuration for each resource is valid.

Under *Workloads*, the OSSMC plugin adds a *Service Mesh* tab that contains the following subtabs:

* *Overview* subtab provides a summary of the selected workload including a localized topology graph showing the workload with all inbound and outbound edges and nodes.
* *Traffic* subtab displays information about all inbound and outbound traffic to the workload.
* *Logs* subtab shows the logs for the workload's containers. 
+
--

> ** You can view container logs individually or in a unified fashion, ordered by log time. This is especially helpful to see how the Envoy sidecar proxy logs relate to your workload's application logs. Y

Reviewer 2: Superfluous "Y"

```suggestion
** You can view container logs individually or in a unified fashion, ordered by log time. This is especially helpful to see how the Envoy sidecar proxy logs relate to your workload's application logs.
```

** You can enable the tracing span integration which then allows you to see which logs correspond to trace spans.
--
+

> * *Metrics* subtab shows both inbound and outbound metric graphs in the corresponding sub-tabs. All the workload metrics can be displayed here, providing you with a detail view of the performance of your workload. 

Reviewer 2: Everywhere else "subtab" is one word - here it is hyphenated, which I assume is a typo.
```suggestion
* *Metrics* subtab shows both inbound and outbound metric graphs in the corresponding subtabs. All the workload metrics can be displayed here, providing you with a detail view of the performance of your workload. 
```

+
--
** You can enable the tracing span integration which allows you to see which spans occurred at the same time as the metrics. Click on a span marker in the graph to view the specific spans associated with that timeframe.
--
+
* *Traces* provides a chart showing the trace spans collected over the given timeframe. 
+
--
** Click on a bubble to drill down into those trace spans; the trace spans can provide you the most low-level detail within your workload application, down to the individual request level. The trace details view will give further details, including heatmaps that provide you with a comparison of one span in relation to other requests and spans in the same timeframe. 

> ** If you hover over a cell in a heatmap, a tooltip will give some details on the cell data

Reviewer 2: Missing end of sentence period.

```suggestion
** If you hover over a cell in a heatmap, a tooltip will give some details on the cell data.
```

////
This CONCEPT module included in the following assemblies:
-service_mesh/v2x/ossm-kiali-ossmc.adoc
////

:_mod-docs-content-type: CONCEPT
[id="ossm-kiali-ossmc-plugin-user-guide_{context}"]
= About the OpenShift Service Mesh Console plugin
//In the title include nouns or noun phrases that are used in the body text.
//Do not start the title of concept modules with a verb..

// Working on converting this to Asciidoc
// Commenting out so there is delineation in this file 
// DESCRIPTION LIST in asciidoc
// -- are an open  block, help keep things together esp with bullets in the middle of stuff
// plugin compatible with OCP 4.15+

[WARNING]
The OpenShift Service Mesh Console (OSSM) plugin only supports a single tenant. Whether that tenant is configured to access only a subset of OpenShift projects or has access cluster-wide to all projects does not matter, however, only a single tenant can be accessed.

> The {SMProductName} (OSSMC) plugin is an extension to the OpenShift Console that provides visibility into your Service Mesh from the OpenShift web console. You can install the OSSMC plugin in only one of two ways: using the OpenShift web console, or through the CLI.

Reviewer 3: ```suggestion
The {SMProductName} (OSSMC) plugin is an extension to the {product-title} web console that provides visibility into your Service Mesh. You can install the OSSMC plugin in only one of two ways: using the OpenShift web console, or through the CLI.
```

> Installing the OSSMC plugin creates a new category, *Service Mesh*, in the main OpenShift web console navigation. Click *Service Mesh* to see:

Reviewer 3: ```suggestion
Installing the OSSMC plugin creates a new category, *Service Mesh*, in the main {product-title} web console navigation. Click *Service Mesh* to see:
```

> 

Reviewer 3: Unless you were explicitly asked to document each of the menu items in the UI, I would advise against it. While technically accurate, it is best practice to focus on user tasks in instead of describing the UI. In your PR description, you mention that end user documentation will be provided elsewhere. And in fact, the user guide already seems to cover this. So will that resource also be made available to users now? Or will you be converting that guide and this content is being provided, until additional information can be converted?

If you must include the following information, I suggest removing it from this module and moving it to a "Using" section.  From a user perspective, I have not installed the plugin yet, but I am now presented with information on all of the elements in the UI.

I am not sure how pressed you are for time, but I thought this was definitely worth mentioning.

Author: Mike and I had a chat. The UI is documented because this plugin also exists in the community and may differ from what Red Hat supports. The documented UI is what Red Hat supports.
